ROMFS/px4fmu_common/init.d/rcS

```diff
@@ -161,8 +163,31 @@ else
 	param select $PARAM_FILE
 	if ! param import
 	then
-		param reset_all
+		echo "ERROR [init] param import failed"
+		set STARTUP_TUNE 2 # tune 2 = ERROR_TUNE
+
+		param dump $PARAM_FILE
+
+		if [ -d "/fs/microsd" ]
+		then
+			dmesg >> /fs/microsd/param_import_fail.txt &
+
+			# try to make a backup copy
+			cp $PARAM_FILE /fs/microsd/param_import_fail.bson &
+		fi
+
+		# try importing from backup file
+		if [ -f "/fs/microsd/parameters_backup.bson" ]
+		then
+			param import /fs/microsd/parameters_backup.bson
+		fi
 	fi
+
+	if [ $SDCARD_AVAILABLE = yes ]
+	then
+		param select-backup /fs/microsd/parameters_backup.bson
+	fi
```

Comment:
Does this always create a backup? How does that work?

Reply:
At the moment it's just a secondary export file. The primary is typically FRAM (/fs/mtd_params), but could also be another file somewhere (eg /fs/microsd/params).